draft-ietf-oauth-sd-jwt-vc.md

```diff
@@ -475,9 +475,157 @@ Additional JWT VC Issuer Metadata configuration parameters MAY also be used.
+object as defined in (#type-metadata-format).
+
+If the claim `vct#integrity` is present in the SD-JWT VC, its value
+`vct#integrity` MUST be an "integrity metadata" string as defined in Section (#document-integrity).
+
+### From a Registry {#retrieval-from-registry}
+
+A Consumer MAY use a registry to retrieve Type Metadata for a type,
+e.g., if the type is not a HTTPS URL or if the Consumer does not have
+access to the URL. The registry MUST be a trusted registry, i.e., the Consumer MUST trust the registry to provide correct Type Metadata for the type.
+
+The registry MUST provide the Type Metadata in the same format as described in
+(#type-metadata-format).
+
+### Using a Defined Retrieval Method {#defined-retrieval-method}
+
+Ecosystems MAY define additional methods for retrieving Type Metadata. For example, a
+standardization body or a community MAY define a service which has to be used to
+retrieve Type Metadata based on a URN in the `vct` claim.
+
+### From a Local Cache {#retrieval-from-local-cache}
+
+A Consumer MAY cache Type metadata for a type. If a hash for integrity
+protection is present in the Type Metadata as defined in (#document-integrity), the Consumer MAY assume that the Type Metadata is static and can be cached
+indefinitely. Otherwise, the Consumer MUST use the `Cache-Control`
+header of the HTTP response to determine how long the metadata can be cached.
+
+### From Type Metadata Glue Documents {#glue-documents}
+
+Credentials MAY encode Type Metadata directly, providing it as "glue
+information" to the consumer.
+
+For JSON-serialized JWS-based credentials, such Type Metadata documents MAY be
+included in the unprotected header of the JWS. In this case, the key `vctm` MUST
+be used in the unprotected header and its value MUST be an array of
+base64url-encoded Type Metadata documents as defined in this specification.
+Multiple documents MAY be included for providing a whole chain of types to the
+Consumer (see (#extending-type-metadata)).
+
+A Consumer of a credential MAY use the documents in the `vctm`
+array instead of retrieving the respective Type Metadata elsewhere as follows:
+
+ * When resolving a `vct` in a credential, the Consumer MUST ensure
+   that the `vct` claim in the credential matches the one in the Type Metadata
+   document, and it MUST verify the integrity of the Type Metadata document as
+   defined in (#document-integrity). The Consumer MUST NOT use the Type Metadata if no hash for integrity protection was provided in `vct#integrity`.
+ * When resolving an `extends` property in a Type Metadata document, the Consumer MUST ensure that the value of the `extends` property in the
+   Type Metadata document matches that of the `vct` in the Type Metadata document, and it MUST verify the integrity of the Type Metadata document as defined in
+   (#document-integrity). The Consumer MUST NOT use the Type Metadata if no hash for integrity protection was provided.
+
+# Document Integrity {#document-integrity}
```

Comment:
This should maybe be a subsection of Type Metadata so... ?
```suggestion
## Document Integrity {#document-integrity}
```

Reply:
It is meant as a mechanism that can be used for multiple claims defined in the spec. If we only need it for the Type Metadata claims, this should be a subsection. Otherwise, a top-level section.

Reply:
It should be a top-level section imo since it is a general mechanism.

Reply:
agreed on the april 2nd call that it should stay a top-level claim.